src/atomate2/abinit/jobs/base.py

```diff
@@ -0,0 +1,364 @@
+
+import jobflow
+from abipy.flowtk.events import as_event_class
+from jobflow import Maker, Response, job
+from monty.json import jsanitize
+from monty.serialization import dumpfn
+from pymatgen.core.structure import Structure
+
+from atomate2.abinit.files import write_abinit_input_set
+from atomate2.abinit.run import run_abinit
+from atomate2.abinit.schemas.core import AbinitTaskDocument, Status
+from atomate2.abinit.sets.base import AbinitInputGenerator, get_extra_abivars
+from atomate2.abinit.utils.common import InitializationError, UnconvergedError
+from atomate2.abinit.utils.history import JobHistory
+from atomate2.settings import Atomate2Settings
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["BaseAbinitMaker"]
+
+
+ResponseArgs = namedtuple(
+    "ResponseArgs",
+    ["detour", "addition", "replace", "stop_children", "stop_jobflow", "stored_data"],
+)
+
+
+JobSetupVars = namedtuple(
+    "JobSetupVars",
+    ["start_time", "history", "workdir", "settings", "abipy_manager", "wall_time"],
+)
+
+
+def setup_job(
+    structure,
+    prev_outputs,
+    restart_from,
+    history,
+    wall_time,
+):
+    # Get the start time.
+    start_time = time.time()
+
+    if structure is None and prev_outputs is None and restart_from is None:
+        raise RuntimeError(
+            "At least one of structure, prev_outputs or "
+            "restart_from should be defined."
+        )
+
+    if history is None:
+        # Supposedly the first time the job is created
+        history = JobHistory()
+    elif restart_from is not None:
+        # We want to log the restart only if the restart_from is due to
+        # an automatic restart, not a restart from e.g. another scf or relax
+        # with different parameters.
+        history.log_restart()
+
+    # Set up working directory
+    workdir = os.getcwd()
+
+    # Log information about the start of the job
+    history.log_start(workdir=workdir, start_time=start_time)
+
+    # Set up logging
+    log_handler = logging.FileHandler("atomate2_abinit.log")
+    log_handler.setFormatter(logging.Formatter(logging.BASIC_FORMAT))
+    logging.getLogger("pymatgen.io.abinit").addHandler(log_handler)
+    logging.getLogger("abipy").addHandler(log_handler)
+    logging.getLogger("atomate2").addHandler(log_handler)
+
+    # Load the atomate settings for abinit to get configuration parameters
+    # TODO: how to allow for tuned parameters on a per-job basis ?
+    #  (similar to fw_spec-passed settings)
+    settings = Atomate2Settings()
+    abipy_manager = None  # Currently disabled as it is needed for autoparal,
+    # which is not yet supported
+    # abipy_manager = get_abipy_manager(settings)
+
+    # set walltime, if possible
+    # TODO: see in set_walltime, where to put this walltime_command
+    wall_time = wall_time
+    return JobSetupVars(
+        start_time=start_time,
+        history=history,
+        workdir=workdir,
+        settings=settings,
```

Comment:
You don't need to pass the settings here. Just use the `SETTINGS` object directly in the make method later on.